server/src/main/java/org/opensearch/action/support/TimeoutTaskCancellationUtility.java

```diff
@@ -100,6 +106,31 @@ public static <Response> ActionListener<Response> wrapWithCancellationListener(
         return listenerToReturn;
     }
 
+    /**
+    * Wraps a listener with a timeout listener {@link TimeoutRunnableListener} to schedule the task cancellation for provided tasks on
+    * generic thread pool
+    * @param task - {@link CancellableTask}
+    * @param listener - original listener associated with the task
+    * @return wrapped listener
+    */
+    public static <T> ActionListener<T> wrapWithCancellationCheck(CancellableTask task, ActionListener<T> listener) {
+        return new ActionListener<T>() {
```

Comment:
comment says  Wraps a listener with a timeout listener {@link TimeoutRunnableListener}  but it is still new ActionListener(), am i missing something?